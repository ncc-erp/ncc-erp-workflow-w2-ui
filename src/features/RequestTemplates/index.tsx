import { useRequestTemplates } from 'api/apiHooks/requestHooks';
import Page from 'common/components/Page';
import { RequestTemplateTable } from './components/RequestTemplateTable';
import { useMemo } from 'react';
import { useUserPermissions } from 'hooks/useUserPermissions';
import { Permissions } from 'common/constants';

const RequestTemplates = () => {
  const { data, isLoading, refetch } = useRequestTemplates();
  const { renderIfAllowed } = useUserPermissions();

  const temp = useMemo(() => {
    return data;
  }, [data]);
  return (
    <>
      {renderIfAllowed(
        Permissions.VIEW_WORKFLOW_DEFINITIONS,
        <Page>
          <Page.Header>
            <Page.HeaderLeft>
              <Page.Heading>Request Templates</Page.Heading>
            </Page.HeaderLeft>
            <Page.HeaderRight />
          </Page.Header>
          <Page.Body>
            <RequestTemplateTable
              data={temp || { items: [], totalCount: 0 }}
              isLoading={isLoading}
              refetch={refetch}
            />
          </Page.Body>
        </Page>
      )}
    </>
  );
};

export default RequestTemplates;
