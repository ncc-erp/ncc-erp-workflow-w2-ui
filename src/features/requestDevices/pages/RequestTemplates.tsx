const RequestTemplates = () => {
  return <div>request template page</div>;
};

export default RequestTemplates;
