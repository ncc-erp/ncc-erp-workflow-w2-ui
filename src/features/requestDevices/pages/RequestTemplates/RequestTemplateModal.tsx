import {
  Modal,
  ModalOverlay,
  ModalContent,
  ModalHeader,
  ModalFooter,
  ModalBody,
  ModalCloseButton,
  Button,
} from '@chakra-ui/react';
interface RequestTemplateModalProps {
  isOpen: boolean;
  onClose: () => void;
  requestId: string;
}
export const RequestTemplateModal = ({
  isOpen,
  onClose,
  requestId,
}: RequestTemplateModalProps) => {
  return (
    <Modal isOpen={isOpen} onClose={onClose}>
      <ModalOverlay />
      <ModalContent>
        <ModalHeader>Modal Title</ModalHeader>
        <ModalCloseButton />
        <ModalBody>This is the content of ID: {requestId}</ModalBody>

        <ModalFooter>
          <Button colorScheme="blue" mr={3} onClick={onClose}>
            Close
          </Button>
          <Button variant="ghost">Secondary Action</Button>
        </ModalFooter>
      </ModalContent>
    </Modal>
  );
};
