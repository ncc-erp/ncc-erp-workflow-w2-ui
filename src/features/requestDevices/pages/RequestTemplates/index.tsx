import { Button } from '@chakra-ui/react';
import { useRequestTemplates } from 'api/apiHooks/requestHooks';
import { useCurrentUser } from 'api/apiHooks/userHooks';
import { useNavigate } from 'react-router-dom';
import { useClearUserData } from 'stores/user';

const RequestTemplates = () => {
  const clearUser = useClearUserData();
  const navigate = useNavigate();
  const { remove } = useCurrentUser();
  const { data: requestTemplateData } = useRequestTemplates();

  const onLogout = () => {
    clearUser();
    navigate('/login');
    remove();
  };

  return (
    <div>
      Request templates {JSON.stringify(requestTemplateData, null, 2)}{' '}
      <Button onClick={onLogout}>Log out</Button>
    </div>
  );
};

export default RequestTemplates;
