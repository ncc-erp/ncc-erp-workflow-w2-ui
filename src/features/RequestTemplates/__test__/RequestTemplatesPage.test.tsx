import { render } from '@testing-library/react';
import { RecoilRoot } from 'recoil';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import RequestTemplates from '..';
import { BrowserRouter as Router } from 'react-router-dom';

jest.mock('../../../api/apiHooks/index', () => ({
  useAxios: jest.fn(),
}));

jest.mock('../../../api/axiosInstant', () => ({
  VITE_API_BASE_URL: '/api',
}));

jest.mock('hooks/useMediaQuery', () => ({
  useMediaQuery: jest.fn().mockReturnValue(true),
}));

jest.mock('api/apiHooks/requestHooks', () => ({
  useRequestTemplates: jest.fn().mockReturnValue({
    isLoading: false,
    data: {
      totalCount: 1,
      items: [
        {
          definitionId: '3a05daee-8f55-1e4c-9ac3-443c04309b77',
          name: 'Change Office Request',
          displayName: 'Change Office Request',
          description: null,
          version: 15,
          isSingleton: false,
          isPublished: true,
          isLatest: true,
          inputDefinition: {
            workflowDefinitionId: '3a05daee-8f55-1e4c-9ac3-443c04309b77',
            propertyDefinitions: [
              {
                name: 'CurrentOffice',
                type: 'OfficeList',
                isRequired: true,
              },
              {
                name: 'DestinationOffice',
                type: 'OfficeList',
                isRequired: true,
              },
              {
                name: 'Content',
                type: 'RichText',
                isRequired: true,
              },
              {
                name: 'StartDate',
                type: 'DateTime',
                isRequired: true,
              },
              {
                name: 'EndDate',
                type: 'DateTime',
                isRequired: false,
              },
            ],
            id: '3a05ffba-3830-e0d4-e931-9381c70a3710',
          },
          id: '3a0e24fb-ebd4-bd23-f2a9-7e447230098e',
        },
      ],
    },
  }),
  useDeleteWorkflowDefinition: jest.fn(() => ({
    mutateAsync: jest.fn(),
  })),
  useUpdateWorkflowPublishStatus: jest.fn(() => ({
    mutateAsync: jest.fn(),
  })),
}));

describe('Request Template Page', () => {
  //  const totalCount = 1;
  test('should match snapshot when rendering', () => {
    const queryClient: QueryClient = new QueryClient();
    const { container } = render(
      <QueryClientProvider client={queryClient}>
        <RecoilRoot>
          <Router>
            <RequestTemplates />
          </Router>
        </RecoilRoot>
      </QueryClientProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
