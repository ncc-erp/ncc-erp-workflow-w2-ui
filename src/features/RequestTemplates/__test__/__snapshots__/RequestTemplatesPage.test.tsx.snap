// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request Template Page 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="chakra-stack css-jrkth6"
    >
      <div
        class="chakra-stack__item css-1stbcqi"
      >
        <h1
          class="chakra-heading css-ftu039"
        >
          Request Templates
        </h1>
      </div>
      <div
        class="chakra-stack__item css-bdvyq0"
      >
        <div>
          <button
            class="chakra-button css-v0af4h"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 160c-52.9 0-96 43.1-96 96s43.1 96 96 96 96-43.1 96-96-43.1-96-96-96zm246.4 80.5l-94.7-47.3 33.5-100.4c4.5-13.6-8.4-26.5-21.9-21.9l-100.4 33.5-47.4-94.8c-6.4-12.8-24.6-12.8-31 0l-47.3 94.7L92.7 70.8c-13.6-4.5-26.5 8.4-21.9 21.9l33.5 100.4-94.7 47.4c-12.8 6.4-12.8 24.6 0 31l94.7 47.3-33.5 100.5c-4.5 13.6 8.4 26.5 21.9 21.9l100.4-33.5 47.3 94.7c6.4 12.8 24.6 12.8 31 0l47.3-94.7 100.4 33.5c13.6 4.5 26.5-8.4 21.9-21.9l-33.5-100.4 94.7-47.3c13-6.5 13-24.7.2-31.1zm-155.9 106c-49.9 49.9-131.1 49.9-181 0-49.9-49.9-49.9-131.1 0-181 49.9-49.9 131.1-49.9 181 0 49.9 49.9 49.9 131.1 0 181z"
              />
            </svg>
            <p
              class="chakra-text css-14gt3bc"
            >
              Light
            </p>
          </button>
        </div>
      </div>
    </div>
    <div
      class="css-0"
    >
      <div
        class="css-0"
      >
        <div
          class="css-1mrxmy5"
        >
          <table
            class="chakra-table css-zy166a"
          >
            <thead
              class="css-0"
            >
              <tr
                class="css-1y39i0r"
              >
                <th
                  class="css-cbljg4"
                  colspan="1"
                >
                  <div
                    class="css-u511cn"
                  >
                    Request Template
                  </div>
                </th>
                <th
                  class="css-cbljg4"
                  colspan="1"
                >
                  <div
                    class="css-u511cn"
                  >
                    <div
                      class="css-mgpx6q"
                    >
                      Actions
                    </div>
                  </div>
                </th>
              </tr>
            </thead>
            <tbody
              class="css-0"
            >
              <tr
                class="css-eg92qj"
              >
                <td
                  class="css-bhmk6k"
                >
                  Change Office Request
                </td>
                <td
                  class="css-bhmk6k"
                >
                  <div
                    class="css-gmuwbf"
                  >
                    <button
                      aria-label="Popup modal"
                      class="chakra-button css-p7mkev"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 24 24"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M11 11V5H13V11H19V13H13V19H11V13H5V11H11Z"
                        />
                      </svg>
                    </button>
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="chakra-stack css-148qy46"
        >
          <div
            class="chakra-stack css-130amzp"
          >
            <div
              class="chakra-stack css-1hohgv6"
            >
              <p
                class="chakra-text css-1bevrez"
              >
                Rows per page
              </p>
              <div
                class="chakra-form-control css-0"
                role="group"
              >
                <div
                  class="chakra-select__wrapper css-13voi5h"
                >
                  <select
                    class="chakra-select css-1y1ernp"
                    id="field-:r0:"
                  >
                    <option
                      value="10"
                    >
                      10
                    </option>
                    <option
                      value="25"
                    >
                      25
                    </option>
                    <option
                      value="50"
                    >
                      50
                    </option>
                    <option
                      value="100"
                    >
                      100
                    </option>
                  </select>
                  <div
                    class="chakra-select__icon-wrapper css-10n8c1x"
                  >
                    <svg
                      aria-hidden="true"
                      class="chakra-icon chakra-select__icon css-13otjrl"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      role="presentation"
                      stroke="currentColor"
                      stroke-width="0"
                      style="color: currentColor; width: 1em; height: 1em;"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12 14L8 10H16L12 14Z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="css-iz7l97"
            />
            <p
              class="chakra-text css-1bevrez"
            >
              showing 1 to 1 of 1 entries
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
